utility_scripts/test/test_threshold_calculator.py

```diff
@@ -0,0 +1,86 @@
+import os
+import unittest
+from pathlib import Path
+
+import pandas as pd
+
+from utility_scripts.threshold_calculator.enums import ExtendedEnum
+from utility_scripts.threshold_calculator.validator import Validator
+from utility_scripts.threshold_calculator.calculator import ThresholdCalculator
+
+_project_root_directory = Path(__file__).absolute().parent.parent.parent
+
+
+class TestExtendedEnum(unittest.TestCase):
+    def test_list_correct(self):
+        class EnumTestCase(ExtendedEnum):
+            SOME_VAR = 'some_var'
+            OTHER_VAR = 'other_var'
+
+        self.assertListEqual(['some_var', 'other_var'], EnumTestCase.list())
+
+
+class TestValidator(unittest.TestCase):
+    depth_1_directory = Path(os.path.join(
+        _project_root_directory,
+        'some_directory'
+    ))
+    depth_2_directory = Path(os.path.join(
+        _project_root_directory,
+        'other_directory',
+        'further_directory'
+    ))
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.validator = Validator()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        os.rmdir(cls.depth_1_directory)
+        os.rmdir(cls.depth_2_directory)
+        os.rmdir(cls.depth_2_directory.parent)
+
+    def test_1_depth_correct(self):
+        self.assertWarns(
```

Comment:
Small suggestion: Might be even better to check the generated error/warning message (especially if the same error/warning could be given by a function but due to different reasons):
```
        with self.assertWarns(UserWarning) as e:
            # call to function that generates warning
        self.assertEqual('expected', str(e.exception)) # str(e.exception) was in a specific IOError for me, e might have a simple variable containing the string in other errors/warnings
```